src/basic/fs-util.c

```diff
@@ -644,6 +644,20 @@ static bool safe_transition(const struct stat *a, const struct stat *b) {
         return a->st_uid == b->st_uid; /* Otherwise we need to stay within the same UID */
 }
 
+static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
+        if (flags & CHASE_WARN) {
+                _cleanup_free_ char *n1, *n2 = NULL;
+
+                (void) fd_get_path(a, &n1);
+                (void) fd_get_path(b, &n2);
+
+                log_error_errno(EPERM, "Unsafe path transition %s → %s in %s, refusing.",
```

Comment:
SYNTHETIC_ERRNO() please, see https://systemd.io/CODING_STYLE

Reply:
Also, the flag is called CHASE_WARN, hence this should be `log_warning_errno()` and not higher (i.e. not at LOG_ERR log level). And I think the `refusing` suffix should be dropped, since the caller might decide to do something else instead. Hence, maybe just do:

```c
log_warning_errno(SYNTHETIC_ERRNO(EPERM), "Detected unsafe symlink transition %s %s %s during canonicalization of %s.", n1, special_glyph(ARROW), n2, path);
```